### PR TITLE
feature: Remote metrics

### LINKED ISSUE #563: Track train/validation metrics in the Sagemaker dashboard when doing remote tuning
**Is your feature request related to a problem? Please describe.**
Sagemaker customers using Renate would like to see metrics being updated in the Sagemaker console page available for each training job. This requires also to aggregate metrics when running a local tuning job with multiple workers within the same training job.

**Describe the solution you'd like**
Training and validation metrics should be displayed in the Sagemaker console (see https://docs.aws.amazon.com/sagemaker/latest/dg/training-metrics.html -- in particular the section "Monitoring Training Job Metrics (SageMaker Console)") 

**Describe alternatives you've considered**
We can discuss this offline.

### PR DESCRIPTION
**Pull request by mseeger**
Closes #563 .

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/tutorials/benchmarking/bm_local.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ locally:
 .. code-block:: bash
 
    python benchmarking/nursery/launch_local/hpo_main.py \
-     --experiment_tag tutorial_local --benchmark resnet_cifar10 \
+     --experiment_tag tutorial-local --benchmark resnet_cifar10 \
```
Comment (mseeger) [inline docs/source/tutorials/benchmarking/bm_local.rst:52]: Fix: Must not contain "_"

**Review by jgolebiowski — CHANGES_REQUESTED** [review]
Looks very good overall. I have a small refactor request to avoid any confusion down the line.

File: benchmarking/commons/launch_remote_local.py [hunk 2]
```diff
@@ -88,6 +89,18 @@ def get_hyperparameters(
...
+def register_metrics_with_estimator(
+    estimator: EstimatorBase, benchmark: RealBenchmarkDefinition
+):
+    callback = RemoteTuningMetricsCallback(
+        metric=benchmark.metric,
+        mode=benchmark.mode,
+        config_space=benchmark.config_space,
+        resource_attr=benchmark.resource_attr,
+    )
+    callback.register_metrics_with_estimator(estimator)
```
[the remark this answers is not in the document]
Comment (mseeger) [reply benchmarking/commons/launch_remote_local.py:101]: You are right, I was taking a shortcut here

File: syne_tune/remote/remote_metrics_callback.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        self.metric_names = [BEST_METRIC_VALUE, BEST_TRIAL_ID]
+        if self._resource_attr is not None:
+            self.metric_names.append(BEST_RESOURCE_VALUE)
+        if config_space is not None:
+            for name, domain in config_space.items():
+                if isinstance(domain, Domain):
+                    self.metric_names.append(BEST_HP_PREFIX + name)
+        if len(self.metric_names) > MAX_METRICS_SUPPORTED_BY_SAGEMAKER:
+            self.metric_names = self.metric_names[:MAX_METRICS_SUPPORTED_BY_SAGEMAKER]
+        logger.info(f"Remote tuning metrics reported: {self.metric_names}")
```
[the remark this answers is not in the document]
Comment (mseeger) [reply syne_tune/remote/remote_metrics_callback.py:86]: Done.

**Review by jgolebiowski — APPROVED** [review]

**Comment by jgolebiowski** [comment]
LGTM! Thats a very cool change.

**Comment by mseeger** [comment]
Giovanni asked for it. It would be better if we had a nice configuration for the CloudWatch dashboard that we could either ask the customer to adjust or (even better) we could auto-adjust as part of calling the training job

**Pull request merged**